src/Process/HostProcessFactory.php

```diff
@@ -62,7 +65,7 @@ private function getDefaultEnv(): array
         return $env;
     }
 
-    public function phpCgi(string $solutionPath, array $env, string $content): Process
+    public function phpCgi(Environment $environment, array $env, string $content): Process
```

Comment:
@mikeymike processFactory is a bit odd for me, because we have cli & cgi runners as separate entities, but then they both depend on processFactory with specific methods for cli and cgi, so it feels like they are all tied together.  Can you think of a signature for the processFactory which would generic?